Userland/Libraries/LibJS/Runtime/Temporal/Duration.cpp

```diff
@@ -422,6 +419,11 @@ ThrowCompletionOr<TimeDurationRecord> balance_duration(VM& vm, double days, doub
 {
     // 1. If relativeTo is not present, set relativeTo to undefined.
 
+    // NOTE: If any of the inputs is not finite this will mean that we have infinities,
+    //       so the duration will never be valid. Also
+    if (!isfinite(days) || !isfinite(hours) || !isfinite(minutes) || !isfinite(seconds) || !isfinite(milliseconds) || !isfinite(microseconds))
+        return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidDuration);
```

Comment:
From a quick look it seems like this still wouldn't solve the problem we have here.

The spec is very loosy goosy about numbers our problem is that if we don't check for infinity at the beginning here we would possibly construct `UnsignedBigInteger`s with infinity.
Clearly there is no correct way of storing infinity as an integer. So we `VERIFY` you don't give the constructors Infinity. However this path without check (for example `days=inf`) lead to `total_duration_nanoseconds` using `days` for BigInteger construction.

One solution would be adding is_infinity, is_nan to BigInteger but that would also requiring touching all operators, algorithm etc. And feels ugly to extend the somewhat weird floating point semantics to BigInteger.

So this feels reasonable since if any input is infinity it will eventually make some value infinity leading to us having to throw the RangeError

What would be your preferred solution?
(Whoops after writing this comment I see I didn't fully understand your comment properly, do you mean we shouldn't do this check?)

Reply:
Just saying the spec already considers that the end result might be infinite and throws in that case, which we don't do before this patch yet - AFAICT. Fine for now, but might need changes based on that new AO later on.

Reply:
Ah got it, yeah this is non-spec so changing it is fine.